File: Data Structures & Algorithms/is-anagram/submission-10.py
class Solution:
	def isAnagram(self, s: str, t:str) -> bool:
		map = {}
		for ch in s:
			if ch in map:
				map[ch]+=1
			else:
				map[ch] = 1
		print(map)
		for ch in t:
			if ch in map:
				map[ch]-=1
			else:
				return False
		print(map)
		for i in map.values():
			if i!=0:
				return False
		return True
	

File: Data Structures & Algorithms/validate-parentheses/submission-10.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack();
        String open = "({[";
        String close = ")}]";
        for(char ch: s.toCharArray()) {
            
            if(open.indexOf(ch)>-1) {
                stack.push(ch);
            } else {
                int index = close.indexOf(ch);
                if (!stack.isEmpty() && open.charAt(index) == stack.peek()) {
                    stack.pop();
                    
                } else {
                    return false;
                }
            }

            // stack.forEach(System.out::println);
        }

        return stack.isEmpty();
    }
}
